### PR TITLE
feature: refactoring

### PR DESCRIPTION
**Pull request by KirillGaliulin**
Доделал рефакторинг из другого открытого (забытого) ПРа https://github.com/abak-press/apress-selenium_eti/pull/28
https://jira.railsc.ru/browse/AT-198

Основная идея:
Разбили один пейдж ЕТИ на отдельные классы по функциональности (например: попапы, которые также раздроблены на классы, Шапка таблицы ЕТИ,  сама таблица ЕТИ, колонка навигации и т.д.)

### CODE REVIEW

File: lib/pages/company_site/eti/popups/groups_binding_popup.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        div(:popup, css: '.groups-popup__title')
+        elements(:groups_tree, css: '.groups-popup__tree li')
+        button(:save, css: '.groups-popup__save-button')
+
+        def wait_for_visible
+          popup_element.when_visible
+        end
+
+        #TODO Надо бы пофиксить, чтобы с другими спеками было идентично
+        #С методом find не проходит. Кажется не видит элементы из списка
```
Comment (KirillGaliulin) [inline lib/pages/company_site/eti/popups/groups_binding_popup.rb:14]: Планирую пофиксить после правок упавших спеков из-за редизайна ЕТИ

File: lib/apress/selenium_eti.rb [hunk 2]
```diff
@@ -1,2 +1,26 @@
+# Упорядоченный рекваир. Небходимо сначала подключать неймспейсы, иначе будут ошибки при инициализации.
```
Comment (apress-hound) [inline lib/apress/selenium_eti.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/mini_eti/pagination.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module CompanySite
```
Comment (apress-hound) [inline lib/pages/company_site/mini_eti/pagination.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/mini_eti/mini_eti.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module CompanySite
```
Comment (apress-hound) [inline lib/pages/company_site/mini_eti/mini_eti.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/eti/table_status_bar.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module CompanySite
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table_status_bar.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/eti/table_products.rb [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+
+          images = images_upload_popup.images
+          images_upload_popup.close
+          images
+        end
+
+        def image(product)
+          {
+            url: images_elements[product_index(product)].attribute('src'),
+            counts: images_counters_elements[product_index(product)].text.to_i
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table_products.rb:435]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/pages/company_site/eti/table_products.rb [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+        end
+
+        def name(product)
+          names_elements[product_index(product)].element.text
+        end
+
+        def battery(product)
+          {
+            level: battery_level_elements[product_index(product)].text.strip.to_i,
+            title: battery_title_elements[product_index(product)].attribute('title')
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table_products.rb:135]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/pages/company_site/eti/table_products.rb [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+            .action
+            .move_to(products_elements[product_index(product)].element)
+            .perform
+
+          copy_product_icon_elements[product_index(product)].click
+          wait_until { products_elements.size > products.size }
+        end
+
+        def delete_product(product)
+
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table_products.rb:92]: Layout/EmptyLinesAroundMethodBody: Extra empty line detected at method body beginning.

File: lib/pages/company_site/eti/table_products.rb [hunk 9]
```diff
@@ -0,0 +1,459 @@
+module CompanySite
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table_products.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/eti/table.rb [hunk 10]
```diff
@@ -0,0 +1,16 @@
+module CompanySite
+  module ETI
+    class Table < Page
+      include CompanySite::ETI
+
+      title(:project_pulscen, xpath: "//title[contains(text(), 'Пульс')]")
+      title(:project_blizko, xpath: "//a[@title='Главная страница Blizko']")
+
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table.rb:8]: Layout/TrailingWhitespace: Trailing whitespace detected.

File: lib/pages/company_site/eti/table.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
+module CompanySite
```
Comment (apress-hound) [inline lib/pages/company_site/eti/table.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/pages/company_site/eti/popups/traits_popup.rb [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        elements(:trait_values, :text_area, css: '.js-popup-traits .js-trait-value')
+        button(:save, css: '.js-popup-traits [title="Сохранить"]')
+        button(:cancel, css: '.js-popup-traits [title="Отменить"]')
+
+        def wait_for_visible
+          popup_element.when_visible
+        end
+
+        def set_trait_value(name, value)
+
```
Comment (apress-hound) [inline lib/pages/company_site/eti/popups/traits_popup.rb:18]: Layout/EmptyLinesAroundMethodBody: Extra empty line detected at method body beginning.

File: lib/pages/company_site/eti/popups/price_popup.rb [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+        def range_price(options = {})
+          select_price_type('range')
+          self.price_value = options[:price] if options[:price]
+          self.price_max_value = options[:price_max] if options[:price_max]
+          self.currency = options[:currency] if options[:currency]
+          self.measure = options[:measure] if options[:measure]
+        end
+
+        def discount_price(options = {})
```
Comment (apress-hound) [inline lib/pages/company_site/eti/popups/price_popup.rb:35]: Metrics/CyclomaticComplexity: Cyclomatic complexity for discount_price is too high. [7/6]

File: lib/pages/company_site/eti/popups/groups_binding_popup.rb [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (apress-hound) [inline lib/pages/company_site/eti/popups/groups_binding_popup.rb:16]: Layout/LeadingCommentSpace: Missing space after #.

File: lib/pages/company_site/eti/popups/groups_binding_popup.rb [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+      class GroupsBindingPopup < self
+        div(:popup, css: '.groups-popup__title')
+        elements(:groups_tree, css: '.groups-popup__tree li')
+        button(:save, css: '.groups-popup__save-button')
+
+        def wait_for_visible
+          popup_element.when_visible
+        end
+
+        #TODO Надо бы пофиксить, чтобы с другими спеками было идентично
```
Comment (apress-hound) [inline lib/pages/company_site/eti/popups/groups_binding_popup.rb:15]: Layout/LeadingCommentSpace: Missing space after #.<br>Style/CommentAnnotation: Annotation keywords like TODO should be all upper case, followed by a colon, and a space, then a note describing the problem.

File: lib/pages/company_site/eti/popups/exists_popup.rb [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+module CompanySite
+  module ETI
+    class Table
+      class ExistsPopup < self
+        EXISTS = {
+          available: 'В наличии',
+          not_available: 'Нет в наличии',
+          order: 'Под заказ',
+          awaiting: 'Ожидается поступление',
+          none: 'Не указано'
```
Comment (apress-hound) [inline lib/pages/company_site/eti/popups/exists_popup.rb:12]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/pages/company_site/eti/navigation_column.rb [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+      def navigate_to_groups
+        rubricator_groups
+        wait_render_table
+      end
+
+      def navigate_to_rubrics
+        rubricator_rubrics
+        wait_render_table
+      end
+
```
Comment (apress-hound) [inline lib/pages/company_site/eti/navigation_column.rb:20]: Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body end.

File: lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+  end
+
+  describe 'Копирование товара' do
+    before do
+      @fields = {
+        name: Faker::Number.number(5),
+        price: {
+          type: :exact,
+          price: Faker::Number.number(5)
+        }
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb:190]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+    end
+  end
+
+  describe 'Копирование товара' do
+    before do
+      @fields = {
+        name: Faker::Number.number(5),
+        price: {
+          type: :exact,
+          price: Faker::Number.number(5)
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb:189]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+      before(:all) do
+        @name = Faker::Number.number(5)
+        @cs_eti_table_products.add_product(name: @name)
+        @product = @cs_eti_table_products.product(name: @name)
+        @battery = @cs_eti_table_products.battery(@product)
+        @discount_price = {
+          type: :discount,
+          price: Faker::Number.number(3),
+          new_price: Faker::Number.number(2),
+          expires_at: Time.now.strftime('%d.%m.%Y')
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb:96]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb [hunk 21]
```diff
@@ -0,0 +1,200 @@
+require 'spec_helper'
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/mini_eti/minieti_spec.rb:1]: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 22]
```diff
@@ -39,88 +42,122 @@
...
-        if @cs_eti_page.project_pulscen?
-          expect(@cs_eti_page.price_value).to eq @price[:wholesale_price].to_s + ' руб. /шт.'
-        elsif @cs_eti_page.project_blizko?
-          expect(@cs_eti_page.price_value).to eq @price[:wholesale_price].to_s + ' ₽ /шт.'
+        if @cs_eti_table.project_pulscen?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("от #{wholesale_price[:price]} руб. /шт. от #{wholesale_price[:min_qty]} шт.")
+        elsif @cs_eti_table.project_blizko?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("от #{wholesale_price[:price]} ₽ /шт. от #{wholesale_price[:min_qty]} шт.")
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:157]: Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 23]
```diff
@@ -39,88 +42,122 @@
...
       end
 
       it 'введеная цена отображается' do
-        if @cs_eti_page.project_pulscen?
-          expect(@cs_eti_page.price_value).to eq @price[:wholesale_price].to_s + ' руб. /шт.'
-        elsif @cs_eti_page.project_blizko?
-          expect(@cs_eti_page.price_value).to eq @price[:wholesale_price].to_s + ' ₽ /шт.'
+        if @cs_eti_table.project_pulscen?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("от #{wholesale_price[:price]} руб. /шт. от #{wholesale_price[:min_qty]} шт.")
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:154]: Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 24]
```diff
@@ -39,88 +42,122 @@
...
-        @price = {wholesale_price: Faker::Number.number(digits: 2), wholesale_number: Faker::Number.number(digits: 1)}
-        @cs_eti_page.set_wholesale_price(@price)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.search_product(@name)
+    context 'когда оптовая цена "от"', story: 'когда оптовая цена "от"' do
+      let(:wholesale_price) do
+        {
+          price: Faker::Number.number(digits: 2),
+          min_qty: Faker::Number.non_zero_digit,
+          not_exact: true
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:147]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 25]
```diff
@@ -39,88 +42,122 @@
...
+        }
+      end
+
+      it 'введеная цена отображается' do
+        if @cs_eti_table.project_pulscen?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("#{wholesale_price[:price]} руб. /шт. от #{wholesale_price[:min_qty]} шт.")
+        elsif @cs_eti_table.project_blizko?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("#{wholesale_price[:price]} ₽ /шт. от #{wholesale_price[:min_qty]} шт.")
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:135]: Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 26]
```diff
@@ -39,88 +42,122 @@
...
+        {
+          price: Faker::Number.number(digits: 2),
+          min_qty: Faker::Number.non_zero_digit
+        }
+      end
+
+      it 'введеная цена отображается' do
+        if @cs_eti_table.project_pulscen?
+          expect(@cs_eti_table_products.wholesale_price(@product))
+          .to eq("#{wholesale_price[:price]} руб. /шт. от #{wholesale_price[:min_qty]} шт.")
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:132]: Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 27]
```diff
@@ -39,88 +42,122 @@
...
+      @cs_eti_table_products.add_product(name: @name, wholesale_price: wholesale_price)
+      @cs_eti_header.search_product(@name, exact: true)
+      @product = @cs_eti_table_products.product(name: @name)
+    end
+
+    context 'когда точная оптовая цена', story: 'когда точная оптовая цена' do
+      let(:wholesale_price) do
+        {
+          price: Faker::Number.number(digits: 2),
+          min_qty: Faker::Number.non_zero_digit
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:125]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 28]
```diff
@@ -39,88 +42,122 @@
...
-        @discount_price = {previous: Faker::Number.number(digits: 3), discount: Faker::Number.number(digits: 2)}
-        @cs_eti_page.set_discount_price(@discount_price)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.search_product(@name)
+      let(:price) do
+        {
+          type: :discount,
+          price: Faker::Number.number(digits: 3),
+          new_price: Faker::Number.number(digits: 2),
+          expires_at: Time.now.strftime('%d.%m.%Y')
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:103]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 29]
```diff
@@ -39,88 +42,122 @@
...
-        before do
-          @price_from_to = {from: Faker::Number.number(digits: 2), to: Faker::Number.number(digits: 3)}
-          @cs_eti_page.set_price_from_to(@price_from_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
+        let(:price) do
+          {
+            type: :range,
+            price: Faker::Number.number(digits: 2),
+            price_max: Faker::Number.number(digits: 3)
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:87]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 30]
```diff
@@ -39,88 +42,122 @@
...
       context 'когда только "до"' do
-        before do
-          @price_to = {to: Faker::Number.number(digits: 2)}
-          @cs_eti_page.set_price_from_to(@price_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
+        let(:price) do
+          {
+            type: :range, 
+            price_max: Faker::Number.number(digits: 2)
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:67]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 31]
```diff
@@ -39,88 +42,122 @@
...
 
       context 'когда только "до"' do
-        before do
-          @price_to = {to: Faker::Number.number(digits: 2)}
-          @cs_eti_page.set_price_from_to(@price_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
+        let(:price) do
+          {
+            type: :range,
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:66]: Layout/TrailingWhitespace: Trailing whitespace detected.

File: lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb [hunk 32]
```diff
@@ -39,88 +42,122 @@
...
     context 'когда цена от и до', story: 'когда цена от и до' do
       context 'когда только "от"' do
-        before do
-          @price_from = {from: Faker::Number.number(digits: 2)}
-          @cs_eti_page.set_price_from_to(@price_from)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
+        let(:price) do
+          {
+            type: :range,
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb:47]: Layout/TrailingWhitespace: Trailing whitespace detected.

File: lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb [hunk 33]
```diff
@@ -2,107 +2,98 @@
...
+          announce: Faker::Movies::Hobbit.quote,
+          description: Faker::Movies::Hobbit.quote,
+          price: {
+            type: :range,
+            price: Faker::Number.number(digits: 3),
+            price_max: Faker::Number.number(digits: 4)
+          },
+          wholesale_price: {
+            price: Faker::Number.number(digits: 2),
+            min_qty: Faker::Number.number(digits: 2)
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb:76]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb [hunk 34]
```diff
@@ -2,107 +2,98 @@
...
+          traits: {
+            CONFIG['eti']['portal_traits']['trait_1'] => CONFIG['eti']['portal_traits']['trait_value_1'],
+            CONFIG['eti']['portal_traits']['trait_2'] => CONFIG['eti']['portal_traits']['trait_value_2']
+          },
+          announce: Faker::Movies::Hobbit.quote,
+          description: Faker::Movies::Hobbit.quote,
+          price: {
+            type: :range,
+            price: Faker::Number.number(digits: 3),
+            price_max: Faker::Number.number(digits: 4)
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb:72]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb [hunk 35]
```diff
@@ -2,107 +2,98 @@
...
-          price_from_to: {from: Faker::Number.number(digits: 3), to: Faker::Number.number(digits: 4)},
-          wholesale_price: {wholesale_price: Faker::Number.number(digits: 2), wholesale_number: Faker::Number.number(digits: 2)}
-        }
-
-        @portal_traits = {
-          trait_1: CONFIG['eti']['portal_traits']['trait_value_1'],
-          trait_2: CONFIG['eti']['portal_traits']['trait_value_2']
+          traits: {
+            CONFIG['eti']['portal_traits']['trait_1'] => CONFIG['eti']['portal_traits']['trait_value_1'],
+            CONFIG['eti']['portal_traits']['trait_2'] => CONFIG['eti']['portal_traits']['trait_value_2']
```
Comment (apress-hound) [inline lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb:65]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: lib/apress/selenium_eti/spec/company_site/eti/product_groups_spec.rb [hunk 36]
```diff
@@ -2,44 +2,52 @@
 
 describe 'ЕТИ' do
   before(:all) do
-    @cs_eti_page = CompanySite::EtiPage.new
-    @cs_main_page = CompanySite::MainPage.new
+    @cs_eti_table                = CompanySite::ETI::Table.new
+    @cs_eti_table                = CompanySite::ETI::Table.new
```
Comment (ovsyanik-ka) [inline lib/apress/selenium_eti/spec/company_site/eti/product_groups_spec.rb:6]: продублировалось

**Review by yaros1avsv — APPROVED** [review]

File: lib/pages/company_site/eti/eti.rb [hunk 37]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module CompanySite
+  module ETI
+    def wait_saving
+      status_bar = CompanySite::ETI::Table::StatusBar.new
+      sleep 1
```
Comment (ovsyanik-ka) [inline lib/pages/company_site/eti/eti.rb:7]: а sleep потмоу что  status_bar не сразу появляется? 
Может сделать wait_until { status_bar? }

File: lib/pages/company_site/eti/eti.rb [hunk 38]
```diff
@@ -0,0 +1,18 @@
...
+      wait_until? { status_bar.status == 'Сохранение изменений...' }
+      wait_until?(10) { status_bar.status == 'Все изменения сохранены' }
+    end
+
+    def wait_render_table
+      table = CompanySite::ETI::Table.new
+      sleep 1
```
Comment (ovsyanik-ka) [inline lib/pages/company_site/eti/eti.rb:14]: тоже может можно заменить sleep?

File: lib/apress/selenium_eti/spec/company_site/eti/product_groups_spec.rb (hunk 36 above)
Comment (KirillGaliulin) [reply lib/apress/selenium_eti/spec/company_site/eti/product_groups_spec.rb:6]: Спасибо, чет не заметил))
поправил

File: lib/pages/company_site/eti/eti.rb (hunk 37 above)
Comment (KirillGaliulin) [reply lib/pages/company_site/eti/eti.rb:7]: Убрал sleep, т.к. далее следует проверка wait_until? { status_bar.status == 'Сохранение изменений...' }

File: lib/pages/company_site/eti/eti.rb (hunk 38 above)
Comment (KirillGaliulin) [reply lib/pages/company_site/eti/eti.rb:14]: Убрал sleep, т.к. далее следует проверка wait_until?(10) { table.load_process_not_exists? }

**Review by ovsyanik-ka — APPROVED** [review]

**Pull request merged**